rampwf/utils/cli/testing.py

```diff
@@ -0,0 +1,81 @@
+import os
+
+import click
+
+from ..testing import assert_notebook
+from ..testing import assert_submission
+
+CONTEXT_SETTINGS = dict(help_option_names=['-h', '--help'])
+
+
+@click.group(context_settings=CONTEXT_SETTINGS)
+def main():
+    """Command-line to test submission and notebook before to submit."""
+    pass
+
+
+@main.command()
+@click.option('--ramp-kit-dir', default='.',
+              help='Root directory of the ramp-kit to test.')
+@click.option('--ramp-data-dir',  default='.',
+              help='Directory containing the data. This directory should '
+              'contain a "data" folder.')
+@click.option('--ramp-submission-dir', default='submissions',
+              help='Directory where the submissions are stored. It should '
+              'contain a "submissions" directory.')
+@click.option('--submission', default='starting_kit',
+              help='The kit to test. It should be located in the '
+              '"submissions" folder of the starting kit. If "ALL", all '
+              'submissions in the directory will be tested.')
+@click.option('--quick-test', is_flag=True,
+              help='Specify this flag to test the submission on a small '
+              'subset of the data.')
+@click.option('--pickle', is_flag=True,
+              help='Specify this flag to pickle the submission after '
+              'training.')
+@click.option('--save-output', is_flag=True,
+              help='Specify this flag to save predictions, scores, eventual '
+              'error trace, and state after training.')
+@click.option('--retrain', is_flag=True,
+              help='Specify this flag to retrain the submission on the full '
+              'training set after the CV loop.')
+def submission(ramp_kit_dir, ramp_data_dir, ramp_submission_dir, submission,
```

Comment:
I would not make this a subcommand of main, but just make this a one-level command?
Because now you would need to do `ramp-test submission --submission name`. We could make it `ramp-test --submission name`, which is much less duplicative.

That would move the testing of the notebook to another command, but I think that is actually a good thing, as no end-user of `rampwf` should need this? (only ramp-kit authors)

Reply:
I don't have objection. We could actually make both testing in the same function and perform whatever testing if you specify notebook, submission or both

Reply:
I agree. But I'd rather do `ramp sdk test_submission --submission`. `test` is a bit too generic for me and it suggests a code test, not an ML model test.